### ISSUE TITLE
Multiple bugs

### ISSUE THREAD

**Issue by isomemo** [Pandora-IsoMemo/iso-app#168]
@arunge testing this file on beta version: 
[upload.xlsx](https://github.com/Pandora-IsoMemo/iso-app/files/11801446/upload.xlsx)

Several things previously working now failing!

1) If I load the data via the Pandora module there are many missing points (compare below map output from "Interactive map" and using "AverageR"

Interactive map

![image](https://github.com/Pandora-IsoMemo/iso-app/assets/46485054/299f46f2-10b5-4d57-b572-1f1c1ad99021)

AverageR

![image](https://github.com/Pandora-IsoMemo/iso-app/assets/46485054/4bea88c5-9b0a-4cd4-80c6-5218cfdc1d51)

2) Has the option to select point colour in the "Interactive map" been made available? I still cannot see it. If I want to set all points to red (instead of default green) how do I do it?

3) Border correction stopped working. If run the AverageR classic model (dependent variable ΔR.14C.yr) and I activate the option "border correction" there is a smooth transition in colours as would be expected. However, this is not working when I select the Bayesian model. This was working before!

Classical model 

![image](https://github.com/Pandora-IsoMemo/iso-app/assets/46485054/f529aee0-06a2-4f5c-b815-260c4f1ef015)

Bayesian model (notice how colours are broken at longitude 180)

![image](https://github.com/Pandora-IsoMemo/iso-app/assets/46485054/f8185602-071e-4a75-acd2-282895d0e686)

4) In AverageR or TimeR if I select as dependent variable "test" I get the following errors:

Using classic model: 'length.out' must be a non-negative number

Using the Bayesian model: 
missing values and NaN's not allowed if 'na.rm' is FALSE

5) If I try to use TimeR (same dependent variable as above and "Date type" option selected "single point" and "Mean or lower time point" selected field "Cal.Age.Cal.BP" there is no modelling output. I tested this same  data some months ago.

It is rather worrying that so many things that were working now have problems.

**Comment by isomemo** [comment]
@arunge You can ignore point 5. Now working! The rest of the points to be checked.

**Comment by arunge** [comment]
@isomemo Some updates regarding the points 1. and 2.  in the interactive map, and poin 4.

- the colour can only be changed in terms of palettes (1). That is, if the colour is fixed, you can still change the palette which will change the colour. I did not implement a different method to colour the points, if "fixed" is selected. Then only the first colour of the palette is used. 

- regarding the "missing points": They are all there. However, leaflet maps do not repeat any data points, only the map itself is repeated. That is, the center seems to be in the atlantic and data is cutted before Australia on the left (2), but after Australia on the right (3).

![image](https://github.com/Pandora-IsoMemo/iso-app/assets/16759098/fd6f1853-4c8f-4ac4-ade5-9bdb6dd8d879)

Regarding point 4.: The column test contains only one value. Here the determination of default ranges for the plot did fail because minima and maxima values of the colour scale are the same. I fixed this by using at least a small range of colours for the legend in the plot. This is fixed.

Regarding points 3. and 5. I will update you soon.

**Comment by isomemo** [comment]
@arunge I believe that in the original specification (and later) I requested the possibility of selecting a specific point colour. Please implement this (even if I did not request it before). The current pallete approach is not practical.

Also not practical to have map points appearing in different "leaflets". One of the goals is to export a map with all points in a centred view. Please fix this.

Point 5 is now working! You can ignore this!

Looking forward to the solution of point 3. This was working before.

**Comment by arunge** [comment]
@isomemo Regarding point 3. there had only been one single change in the relevant part of the code since we shifted all code to our recent repository:

https://github.com/Pandora-IsoMemo/iso-app/blame/043ab7e4beeec59a0f6f91ff647b4faea37ff1aa/R/01-estimateMap.R#L149-L275

I checked a version from before that change (may 2022) from our [archive](https://github.com/Pandora-IsoMemo/drat/tree/main/docs/src/contrib/Archive/MpiIsoApp). Even there the border correction is not working for the Bayes option:

![image](https://github.com/Pandora-IsoMemo/iso-app/assets/16759098/c6c848c4-9e0a-46ed-89be-d35358556f09)

So it was at least not working since the app is open source. The easiest here would be to ask @mgross to check why the border correction is not working.

**Comment by isomemo** [comment]
@arunge I have no idea about the repository changes and why code would be lost in these transfers. Please fix it so that it works as before. Btw check also if the spherical modelling is working.

**Comment by arunge** [comment]
Of course, no code was lost in the transfer. We just copied from one repository to another one. I just wanted to say that this issue must be a very old one that has nothing to do with recent changes.

**Comment by arunge** [comment]
@isomemo As I just said, **nothing** was lost during the transfer! We copied **all** existing code into new repositories.
With "old" I mean the bug was already present in the version that was copied from the previous repository, that means border correction was already not working for Bayes models before January 2022 (the time when we copied all apps to their new repositories). I will ask @mgross to check why the border correction cannot be seen in the plot.

**Comment by isomemo** [comment]
@arunge but in that case it comes back to the issue of tracking impacts on software when changes are made (perhaps a package update?). Ask Marcus to also check the spherical modelling as this may have also been impacted.

**Comment by arunge** [comment]
I opened new issued for the open tasks:

- https://github.com/Pandora-IsoMemo/iso-app/issues/175
- https://github.com/Pandora-IsoMemo/iso-app/issues/174
- https://github.com/Pandora-IsoMemo/iso-app/issues/173

So we can close here
